src/main/java/dev/openfeature/javasdk/Value.java

```diff
@@ -22,6 +22,10 @@ public Value() {
         this.innerObject = null; 
     }
 
+    public Value(Object value) {
+        this.innerObject = value; 
```

Comment:
I am a  bit weary of this.

It effectively removes the constraints on the values in lists. Because that is an external way a value can be accepted directly. I suppose provider wise I could just ignore values that don't comply to any of the `is` checks.

Reply:
> I am a bit weary of this.
> 
> It effectively removes the constraints on the values in lists. Because that is an external way a value can be accepted directly. I suppose provider wise I could just ignore values that don't comply to any of the `is` checks.

I think you mean "wary", but point taken!

We could throw if the object is not one of the valid types (but I'm a little worried about the consequences of that) - or we could make the inner value null in that case. I'm not sure either of these is a great solution.

@rgrassian-split specifically requested this to make his provider simpler... but he's doing the type of check you're suggesting. Maybe this change removes some guardrails we actually want. I'm open to further discussion and input from @rgrassian-split as well.

Reply:
It could maybe complain loudly if it doesn't pass one of the `is` checks.

Reply:
> The constructor could maybe complain loudly if it doesn't pass one of the is checks.

If we throw a checked exception if the object is not a valid type, we will have safety and flexibility, I think. It will just require the called to use a try/catch construct. I personally wouldn't use this in my provider, but I think that might solve @rgrassian-split 's use case without causing issues down the line as @kinyoklion is validly concerned about.

What do you guys think?

Reply:
I think that makes sense.

Reply:
OK. I've done so: https://github.com/open-feature/java-sdk/pull/65/commits/96623191d05937dfabe1e6cef9e4c03494ddca8b

Reply:
Yeah I agree the type checking should still happen and am totally fine handling a case where the type doesn't match